src/Tasks/TaskRequiresFramework.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if !NETFRAMEWORK
+
+using System;
+
+namespace Microsoft.Build.Tasks
+{
+    public abstract class TaskRequiresFramework : TaskExtension
```

Comment:
This being public but `!NETFRAMEWORK` is causing an alert

```xml
  <Suppression>
    <DiagnosticId>CP0001</DiagnosticId>
    <Target>T:Microsoft.Build.Tasks.TaskRequiresFramework</Target>
    <Left>ref/netstandard2.0/Microsoft.Build.Tasks.Core.dll</Left>
    <Right>ref/net472/Microsoft.Build.Tasks.Core.dll</Right>
  </Suppression>
```

I'm not _completely_ opposed to allowlisting this, but we could also make it a no-op `[Obsolete]` type in framework and avoid the error. @jrdodds got a strong opinion?

Reply:
I don't have a strong opinion either way. I'm fine with the no-op [Obsolete] type approach.